### PR TITLE
[Xorg_xkeyboard_config] Initialize XKB_CONFIG_ROOT

### PR DESCRIPTION
**Pull request by barche**

### CODE REVIEW

**Comment by barche** [comment]
This should make sure downstream packages such as Gtk.jl don't need to set this. See https://github.com/JuliaGraphics/Gtk.jl/pull/470

File: X/Xorg_xkeyboard_config/build_tarballs.jl [hunk 1]
```diff
@@ -34,5 +35,9 @@ dependencies = [
     Dependency("Xorg_xkbcomp_jll"),
 ]
 
+init_block = raw"""
+ENV["XKB_CONFIG_ROOT"] = get(ENV, "XKB_CONFIG_ROOT", joinpath(artifact_dir, "share", "X11", "xkb"))
```
Comment (giordano) [inline X/Xorg_xkeyboard_config/build_tarballs.jl:39]: Need `if Sys.islinux() || Sys.isfreebsd()`?  Especially now that this package became `AnyPlatform`

Comment (barche) [reply X/Xorg_xkeyboard_config/build_tarballs.jl]: OK, done!

File: X/Xorg_xkeyboard_config/build_tarballs.jl [hunk 2]
```diff
@@ -34,5 +35,11 @@ dependencies = [
...
 
+init_block = raw"""
+if Sys.islinux() || Sys.isfreebsd()
+    ENV["XKB_CONFIG_ROOT"] = get(ENV, "XKB_CONFIG_ROOT", joinpath(artifact_dir, "share", "X11", "xkb"))
+end
+"""
+
 # Build the tarballs.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; init_block)
```
Comment (giordano) [inline X/Xorg_xkeyboard_config/build_tarballs.jl:45]: AnyPlatform doesn't require changing Julia compat

Comment (giordano) [reply X/Xorg_xkeyboard_config/build_tarballs.jl:45]: Ah, sorry, misread the line!

Comment (giordano) [reply X/Xorg_xkeyboard_config/build_tarballs.jl:45]: Although there's actually a benefit in requiring Julia 1.6: we can safely drop Pkg from the dependencies of the package.

Comment (barche) [reply X/Xorg_xkeyboard_config/build_tarballs.jl:45]: Good to merge I think?

Comment (barche) [reply X/Xorg_xkeyboard_config/build_tarballs.jl:45]: ah, missed that last comment. Want me to add Julia 1.6 compat and remove the Pkg dep?

Comment (giordano) [reply X/Xorg_xkeyboard_config/build_tarballs.jl]: Yeah, even if not strictly required, dropping Pkg from dependencies is a good thing

**Pull request merged**